src/cursor/cur_oligarch.c

```diff
@@ -1265,6 +1289,53 @@ __wt_coligarch_close(WT_CURSOR *cursor)
     API_END_RET(session, ret);
 }
 
+/*
+ * __coligarch_next_random --
+ *     WT_CURSOR->next_random method for the oligarch cursor type.
+ */
+static int
+__coligarch_next_random(WT_CURSOR *cursor)
```

Comment:
This looks good for now, but it would be better if the random cursor would also consider the ingest table—not just when the stable table is empty. Could you please add a TODO?